File: 4-ISP/ISP-Solucao/CadastroCliente.php
<?php
require_once('Interfaces/ICadastroCliente.php');

class CadastroCliente implements ICadastroCliente
{
    public function validarDados()
    {
        // Validar CPF, Email
        echo "\n validar dados";
    }

    public function salvarBanco()
    {
        // Insert na tabela Cliente
        echo "\n salvar banco";
    }

    public function enviarEmail()
    {
        // Enviar e-mail para o cliente
        echo "\n enviar Email";
    }
}

File: Python/4-ISP/ISP-Solucao/CadastroCliente.py
from Interfaces.ICadastroCliente import ICadastroCliente

class CadastroCliente(ICadastroCliente):
    def validarDados(self):
        print('Validar dados')
    
    def salvarBanco(self):
        print('Salvar banco')
    
    def enviarEmail(self):
        print('Enviar e-mail')